File: test/fixtures/java/Advanced.java
/**
 * 高级语法演示：record / 泛型 / 接口默认方法 / 可变参数
 */
public record Point(int x, int y) {
    /** 计算距离 */
    public double distance() {
        return Math.sqrt(x * x + y * y);
    }
}

/**
 * 泛型容器
 */
public class Box<T> {
    /** 内部值 */
    private T value;

    /** 存储值 */
    public void setValue(T value) {
        this.value = value;
    }

    /** 获取值 */
    public T getValue() {
        return value;
    }

    /** 可变参数求和 */
    public static int sum(int... numbers) {
        int total = 0;
        for (int n : numbers) total += n;
        return total;
    }
}

/**
 * 处理器接口
 */
public interface Processor {
    /** 默认处理 */
    default String process(String input) {
        return input;
    }
}

File: test/fixtures/csharp/Advanced.cs
/**
 * 泛型列表
 */
public class ListBox<T>
{
    /** 元素集合 */
    private List<T> _items = new();

    /** 添加元素 */
    public void Add(T item)
    {
        _items.Add(item);
    }

    /** 移除元素 */
    public bool Remove(T item)
    {
        return _items.Remove(item);
    }
}

/**
 * 点记录
 */
public record Point(int X, int Y);

/**
 * 静态计算器
 */
public static class Calculator
{
    /** 静态累加 */
    public static int Total { get; set; }

    /** 累加方法 */
    public static void Add(int value)
    {
        Total += value;
    }
}
